packages/extension/src/store/synchnorousCookieStore.ts

```diff
@@ -323,8 +357,133 @@ class SynchnorousCookieStore {
       devToolsOpenState: false,
       popupOpenState: false,
       newUpdates: 0,
-      portRef: null,
+      frameIDURLSet: {},
+      parentChildFrameAssociation: {},
     };
+    (async () => {
+      if (!this.cookieDB) {
+        this.cookieDB = await fetchDictionary();
+      }
+    })();
+  }
+
+  /**
+   * Clears the whole storage.
+   */
+  clear() {
+    Object.keys(this.tabsData).forEach((key) => {
+      delete this.tabsData[Number(key)];
+    });
+    Object.keys(this.tabs).forEach((key) => {
+      delete this.tabs[Number(key)];
+    });
+    this.tabsData = {};
+    this.tabs = {};
+  }
+
+  /**
+   * This function will deinitialise variables for given tab.
+   * @param {string} tabId The tab whose data has to be deinitialised.
+   */
+  deinitialiseVariablesForTab(tabId: string) {
+    delete this.unParsedRequestHeaders[tabId];
+    delete this.unParsedResponseHeaders[tabId];
+    delete this.requestIdToCDPURLMapping[tabId];
+    delete this.frameIdToResourceMap[tabId];
+  }
+
+  /**
+   * This function will deinitialise variables for given tab.
```

Comment:
Nit: Seems wrong comment for the callback.

Reply:
Fixed in latest commits.